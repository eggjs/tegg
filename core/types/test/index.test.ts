import snapshot from 'snap-shot-it';
import * as types from '../src/index.js';

describe('test/index.test.ts', () => {
  it('should export stable', async () => {
    snapshot(types);
  });
});
