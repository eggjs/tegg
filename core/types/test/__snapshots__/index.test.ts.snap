// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should export stable 1`] = `
{
  "ASPECT_LIST": Symbol(EggPrototype#aspectList),
  "AccessLevel": {
    "PRIVATE": "PRIVATE",
    "PUBLIC": "PUBLIC",
  },
  "CONSTRUCTOR_QUALIFIER_META_DATA": Symbol(EggPrototype#constructorQualifier),
  "CONTROLLER_ACL": Symbol(EggPrototype#controller#acl),
  "CONTROLLER_AOP_MIDDLEWARES": Symbol(EggPrototype#controller#aopMiddlewares),
  "CONTROLLER_HOST": Symbol(EggPrototype#controllerHost),
  "CONTROLLER_HTTP_PATH": Symbol(EggPrototype#controller#http#path),
  "CONTROLLER_META_DATA": Symbol(EggPrototype#controller#metaData),
  "CONTROLLER_METHOD_METHOD_MAP": Symbol(EggPrototype#controller#method#http#method),
  "CONTROLLER_METHOD_PARAM_NAME_MAP": Symbol(EggPrototype#controller#method#http#params#name),
  "CONTROLLER_METHOD_PARAM_TYPE_MAP": Symbol(EggPrototype#controller#method#http#params#type),
  "CONTROLLER_METHOD_PATH_MAP": Symbol(EggPrototype#controller#method#http#path),
  "CONTROLLER_METHOD_PRIORITY": Symbol(EggPrototype#controller#method#http#priority),
  "CONTROLLER_MIDDLEWARES": Symbol(EggPrototype#controller#middlewares),
  "CONTROLLER_NAME": Symbol(EggPrototype#controllerName),
  "CONTROLLER_TYPE": Symbol(EggPrototype#controllerType),
  "CROSSCUT_INFO_LIST": Symbol(EggPrototype#crosscutInfoList),
  "ColumnFormat": {
    "DEFAULT": "DEFAULT",
    "DYNAMIC": "DYNAMIC",
    "FIXED": "FIXED",
  },
  "ColumnType": {
    "BIGINT": "BIGINT",
    "BINARY": "BINARY",
    "BIT": "BIT",
    "BLOB": "BLOB",
    "BOOL": "BOOL",
    "CHAR": "CHAR",
    "DATE": "DATE",
    "DATETIME": "DATETIME",
    "DECIMAL": "DECIMAL",
    "DOUBLE": "DOUBLE",
    "ENUM": "ENUM",
    "FLOAT": "FLOAT",
    "GEOMETRY": "GEOMETRY",
    "GEOMETRYCOLLECTION": "GEOMETRYCOLLECTION",
    "INT": "INT",
    "JSON": "JSON",
    "LINESTRING": "LINESTRING",
    "LONGBLOB": "LONGBLOB",
    "LONGTEXT": "LONGTEXT",
    "MEDIUMBLOB": "MEDIUMBLOB",
    "MEDIUMINT": "MEDIUMINT",
    "MEDIUMTEXT": "MEDIUMTEXT",
    "MULTILINESTRING": "MULTILINESTRING",
    "MULTIPOINT": "MULTIPOINT",
    "MULTIPOLYGON": "MULTIPOLYGON",
    "POINT": "POINT",
    "POLYGON": "POLYGON",
    "SET": "SET",
    "SMALLINT": "SMALLINT",
    "TEXT": "TEXT",
    "TIME": "TIME",
    "TIMESTAMP": "TIMESTAMP",
    "TINYBLOB": "TINYBLOB",
    "TINYINT": "TINYINT",
    "TINYTEXT": "TINYTEXT",
    "VARBINARY": "VARBINARY",
    "VARCHAR": "VARCHAR",
    "YEAR": "YEAR",
  },
  "CompressionType": {
    "LZ4": "LZ4",
    "NONE": "NONE",
    "ZLIB": "ZLIB",
  },
  "ConfigSourceQualifierAttribute": Symbol(Qualifier.ConfigSource),
  "ControllerType": {
    "HEADERS": "HEADERS",
    "HTTP": "HTTP",
    "MESSAGE": "MESSAGE",
    "MGW_RPC": "MGW_RPC",
    "MGW_RPC_STREAM": "MGW_RPC_STREAM",
    "SCHEDULE": "SCHEDULE",
    "SOFA_RPC": "SOFA_RPC",
    "SOFA_RPC_STREAM": "SOFA_RPC_STREAM",
  },
  "DAL_COLUMN_INFO_MAP": Symbol(EggPrototype#dalColumnInfoMap),
  "DAL_COLUMN_TYPE_MAP": Symbol(EggPrototype#dalColumnTypeMap),
  "DAL_INDEX_LIST": Symbol(EggPrototype#dalIndexList),
  "DAL_IS_DAO": Symbol(EggPrototype#dalIsDao),
  "DAL_IS_TABLE": Symbol(EggPrototype#dalIsTable),
  "DAL_TABLE_PARAMS": Symbol(EggPrototype#dalTableParams),
  "DEFAULT_PROTO_IMPL_TYPE": "DEFAULT",
  "DataSourceInjectName": "dataSource",
  "DataSourceQualifierAttribute": Symbol(Qualifier.DataSource),
  "EggLoadUnitType": {
    "APP": "APP",
    "MODULE": "MODULE",
    "PLUGIN": "PLUGIN",
  },
  "EggObjectStatus": {
    "DESTROYED": "DESTROYED",
    "DESTROYING": "DESTROYING",
    "ERROR": "ERROR",
    "PENDING": "PENDING",
    "READY": "READY",
  },
  "EggQualifierAttribute": Symbol(Qualifier.Egg),
  "EggType": {
    "APP": "APP",
    "CONTEXT": "CONTEXT",
  },
  "ErrorCodes": {
    "EGG_PROTO_NOT_FOUND": "EGG_PROTO_NOT_FOUND",
    "INCOMPATIBLE_PROTO_INJECT": "INCOMPATIBLE_PROTO_INJECT",
    "MULTI_PROTO_FOUND": "MULTI_PROTO_FOUND",
  },
  "HTTPMethodEnum": {
    "DELETE": "DELETE",
    "GET": "GET",
    "HEAD": "HEAD",
    "OPTIONS": "OPTIONS",
    "PATCH": "PATCH",
    "POST": "POST",
    "PUT": "PUT",
  },
  "HTTPParamType": {
    "BODY": "BODY",
    "COOKIES": "COOKIES",
    "HEADERS": "HEADERS",
    "PARAM": "PARAM",
    "QUERIES": "QUERIES",
    "QUERY": "QUERY",
    "REQUEST": "REQUEST",
  },
  "INIT_TYPE_TRY_ORDER": [
    "CONTEXT",
    "SINGLETON",
    "ALWAYS_NEW",
  ],
  "IS_CROSSCUT_ADVICE": Symbol(EggPrototype#isCrosscutAdvice),
  "IS_MODEL": Symbol(EggPrototype#model#isModel),
  "IS_SCHEDULE": Symbol(EggPrototype#isSchedule),
  "IS_TRANSACTION_CLAZZ": Symbol(EggPrototype#IS_TRANSACTION_CLAZZ),
  "IndexStoreType": {
    "BTREE": "BTREE",
    "HASH": "HASH",
  },
  "IndexType": {
    "FULLTEXT": "FULLTEXT",
    "INDEX": "INDEX",
    "PRIMARY": "PRIMARY",
    "SPATIAL": "SPATIAL",
    "UNIQUE": "UNIQUE",
  },
  "InitTypeQualifierAttribute": Symbol(Qualifier.InitType),
  "InjectType": {
    "CONSTRUCTOR": "CONSTRUCTOR",
    "PROPERTY": "PROPERTY",
  },
  "InsertMethod": {
    "FIRST": "FIRST",
    "LAST": "LAST",
    "NO": "NO",
  },
  "LoadUnitNameQualifierAttribute": Symbol(Qualifier.LoadUnitName),
  "METHOD_ACL": Symbol(EggPrototype#method#acl),
  "METHOD_AOP_MIDDLEWARES": Symbol(EggPrototype#method#aopMiddlewares),
  "METHOD_AOP_REGISTER_MAP": Symbol(EggPrototype#method#aopMiddlewaresRegister),
  "METHOD_CONTEXT_INDEX": Symbol(EggPrototype#controller#method#context),
  "METHOD_CONTROLLER_HOST": Symbol(EggPrototype#controller#mthods#host),
  "METHOD_CONTROLLER_TYPE_MAP": Symbol(EggPrototype#controller#mthods),
  "METHOD_MIDDLEWARES": Symbol(EggPrototype#method#middlewares),
  "MODEL_DATA_ATTRIBUTES": Symbol(EggPrototype#model#attributes),
  "MODEL_DATA_INDICES": Symbol(EggPrototype#model#indices),
  "MODEL_DATA_SOURCE": Symbol(EggPrototype#model#dataSource),
  "MODEL_DATA_TABLE_NAME": Symbol(EggPrototype#model#tableName),
  "MODEL_PROTO_IMPL_TYPE": "MODEL_PROTO",
  "MethodType": {
    "HTTP": "HTTP",
    "MESSAGE": "MESSAGE",
    "MGW_RPC": "MGW_RPC",
    "MGW_RPC_STREAM": "MGW_RPC_STREAM",
    "SCHEDULE": "SCHEDULE",
    "SOFA_RPC": "SOFA_RPC",
    "SOFA_RPC_STREAM": "SOFA_RPC_STREAM",
  },
  "MultiInstanceType": {
    "DYNAMIC": "DYNAMIC",
    "STATIC": "STATIC",
  },
  "ObjectInitType": {
    "ALWAYS_NEW": "ALWAYS_NEW",
    "CONTEXT": "CONTEXT",
    "SINGLETON": "SINGLETON",
  },
  "POINTCUT_ADVICE_INFO_LIAR": Symbol(EggPrototype#pointcutAdviceInfoList),
  "PROPERTY_QUALIFIER_META_DATA": Symbol(EggPrototype#propertyQualifier),
  "PointcutType": {
    "CLASS": "CLASS",
    "CUSTOM": "CUSTOM",
    "NAME": "NAME",
  },
  "PropagationType": {
    "ALWAYS_NEW": "ALWAYS_NEW",
    "REQUIRED": "REQUIRED",
  },
  "QUALIFIER_IMPL_MAP": Symbol(EggPrototype#qualifierImplMap),
  "QUALIFIER_META_DATA": Symbol(EggPrototype#qualifier),
  "RowFormat": {
    "COMPACT": "COMPACT",
    "COMPRESSED": "COMPRESSED",
    "DEFAULT": "DEFAULT",
    "DYNAMIC": "DYNAMIC",
    "FIXED": "FIXED",
    "REDUNDANT": "REDUNDANT",
  },
  "SCHEDULE_METADATA": Symbol(EggPrototype#schedule#metadata),
  "SCHEDULE_OPTIONS": Symbol(EggPrototype#schedule#options),
  "SCHEDULE_PARAMS": Symbol(EggPrototype#schedule#params),
  "ScheduleType": {
    "ALL": "all",
    "WORKER": "worker",
  },
  "SqlType": {
    "BLOCK": "BLOCK",
    "DELETE": "DELETE",
    "INSERT": "INSERT",
    "SELECT": "SELECT",
    "UPDATE": "UPDATE",
  },
  "TRANSACTION_META_DATA": Symbol(EggPrototype#transaction#metaData),
  "Templates": {
    "BASE_DAO": "base_dao",
    "DAO": "dao",
    "EXTENSION": "extension",
  },
}
`;
